subsys/bluetooth/services/ots/ots.c

```diff
@@ -78,6 +78,59 @@ static ssize_t ots_obj_name_read(struct bt_conn *conn,
 				 strlen(ots->cur_obj->metadata.name));
 }
 
+#if CONFIG_BT_OTS_OBJ_NAME_WRITE_SUPPORT
+ssize_t ots_obj_name_write(struct bt_conn *conn,
+			   const struct bt_gatt_attr *attr,
+			   const void *buf, uint16_t len,
+			   uint16_t offset, uint8_t flags)
+{
+	struct bt_ots *ots = (struct bt_ots *) attr->user_data;
+	struct bt_gatt_ots_object *obj = NULL;
+	int rc = 0;
+	char name[CONFIG_BT_OTS_OBJ_MAX_NAME_LEN + 1];
```

Comment:
Does the RX thread stack have enough for this?

Reply:
I haven't run into any stack overrun issues but I also verified that after the allocation of the name the stack pointer is:
```
(gdb) p $sp
$3 = (void *) 0x20004890 <recv_thread_stack+560>
```
So looks like we still have a fair bit of margin.

Reply:
Thanks for checking.

You can use the thread analyzer: https://docs.zephyrproject.org/2.6.0/guides/flash_debug/thread-analyzer.html to check thread stack usage in an easy way.

Reply:
Thanks for that! Much easier and more accurate way to do this analysis:
```
Thread analyze:
 BT RX               : STACK: unused 160 usage 864 / 1024 (84 %); CPU: 0 %
 BT RX pri           : STACK: unused 256 usage 192 / 448 (42 %); CPU: 0 %
 BT ECC              : STACK: unused 256 usage 888 / 1144 (77 %); CPU: 0 %
 BT TX               : STACK: unused 288 usage 352 / 640 (55 %); CPU: 0 %
 thread_analyzer     : STACK: unused 640 usage 384 / 1024 (37 %); CPU: 1 %
 sysworkq            : STACK: unused 312 usage 712 / 1024 (69 %); CPU: 0 %
 logging             : STACK: unused 664 usage 104 / 768 (13 %); CPU: 0 %
 idle 00             : STACK: unused 264 usage 56 / 320 (17 %); CPU: 98 %
```